tree/30_generic_methods/logger_rudder.cf

```diff
@@ -50,9 +52,31 @@ bundle agent logger_rudder(message, class_prefix)
         int        => getfields("^${current_technique_report_info.technique_name};;${c_class_prefix};;.*", "${expected_reports_file}", ";;", "report_data");
 
   classes:
-      "report_data_read" expression => isgreaterthan("${number_lines}","0");
+      "report_data_read"    expression => isgreaterthan("${number_lines}","0");
+  
+      # We want to detect if expected report data is available in v2 format.
+      # V2 format is characterized by a new field for a total of 6 fields.
+      # The file is rewritten first in .tmp file then in .res file
+      # The second rewrite is done using the edit_line below, so if the 6th field doesn't exist in the original file
+      # then it is translater to a litteral "$reports[0][5]}" in the file.
+      # If there has been no rewrite at all, then there is no 6th field at all in the final file
 
-      "keys_defined"     expression => isvariable("keys");
+      # 1/ The file has been rewritten with an empty field
+      # 5 vs 6 : because getfields and readstringarrayidx do not have the same indexing method (getfields starts at 1)
+      "expected_report_v2_csv"     not => strcmp("${report_data[6]}","${const.dollar}{reports[0][5]}");
+
+      # 2/ The file has never been rewritten
+      "expected_report_v2_res"     not => strcmp("${report_data[6]}", "${const.dollar}{report_data[6]}");
+
+      # 3/ We add another condition : the method promiser must have been rewritten in the technique (the old value is always method_call)
+      "method_v2"                  not => strcmp("method_call", nth("this.callers_promisers", "1"));
```

Comment:
This comment doesn't explain the source of the method_call magic value
In the technique generated with the technique builder - old version, the promiser used is "method_call"